velox/expression/tests/ExprTest.cpp

```diff
@@ -2980,3 +2983,25 @@ TEST_F(ExprTest, addNulls) {
     checkResult(slicedVector);
   }
 }
+
+TEST_F(ExprTest, mapKeysAndValues) {
+  // Verify that the right size of maps and keys arrays are created. This is
+  // done by partially materializing a lazy map vector so that the size of its
+  // underlying vectors is smaller than the rows.size(). Finally if
+  // map_keys/_values tried to create a larger result array, this will throw.
+  VectorFuzzer::Options opts;
```

Comment:
Tests using Fuzzer are a bit opaque. I wonder if this issue can be reproduced using hard-coded data?

```
TEST_F(ExprTest, repro) {
  auto offsets = makeIndices(50, [](auto row) { return 0; });
  auto sizes = makeIndices(50, [](auto row) { return 0; });
  VectorPtr mapVector = std::make_shared<MapVector>(
      pool_.get(),
      MAP(BIGINT(), BIGINT()),
      makeNulls(100, [](auto row) { return row > 50; }),
      100,
      offsets,
      sizes,
      makeFlatVector<int64_t>({1, 2, 3}),
      makeFlatVector<int64_t>({10, 20, 30}));

  evaluate("map_keys(c0)", makeRowVector({mapVector}));
  evaluate("map_values(c0)", makeRowVector({mapVector}));
}
```

Reply:
Unfortunately the test case that you shared above will hit the consistency check in #3078 while creating the mapVector (offset.size() < 100).
Therefore to hit the check inside map_keys/values instead we need the generate a map smaller than rows.size() during evaluation, which I achieved by wrapping a map vector in lazy then materializing (Size-1) part of it.

However, while writing this reply I just realized I can achieve the same by creating a map of size 100 and a SelectivityVector of size 101 (marking the 100th idx as invalid) then executing the eval. thereby avoiding all the fuzzer + lazy hassle. So, I'll post the simplified version soon